### ISSUE TITLE
Safer pacaur script (shellcheck analysis)

### ISSUE THREAD

**Issue by Zopieux** [rmarquis/pacaur#315]
Hello,

This is somehow related to #175. I ran `pacaur` through ShellCheck and it found quite a few hints to make the script safer (mainly quote escapes). Because ShellCheck is not right about everything, it will need some fine tuning to actually extract the useful advises.

Here is the analysis result from master/HEAD (cbac3dd9a96b9cb99ce421782e74dbe3a9c20200): https://gist.github.com/Zopieux/80e810f99b3fbd7160c7

I suggest progressively implementing the useful advises in future releases.

**Comment by rmarquis** [comment]
Hi,

Thanks a lot for the suggestion! I actually closed #175 a long time ago, and the reason it was closed isn't valid anymore, as pacaur now fully rely on the AUR RPC interface instead of bash parsing. I'll have a look at your analysis soon. Any patch is also warmly welcome!

**Comment by Zopieux** [comment]
Thanks for you quick answer. I would be more than willing to contribute a PR, the thing is, I wouldn't want to introduce bugs by blindly applying SpellChecker hints, because I don't have a deep knowledge of bash tricks and pacaur is quite big. I guess some of the most obvious ones can be safely applied though.

I'll have a look when I find the time. Cheers!

**Comment by rmarquis** [comment]
I had a quick look, and you're right, it might be difficult for a third party to improve the code without the required insight. For examples, some of the warnings are actually false-positive (f.e. the "Don't quote rhs of =~, it'll match literally" are done on purpose, or some of the "array expansion" warnings are irrelevant since package names never have space in them). Maybe there is a better way to achieve what I wanted in bash here.

Anyway, I'll have a close look at these warning and improve what should be improved. Thanks again!

**Comment by Zopieux** [comment]
>package names never have space in them

I am not sure I would refrain from adding some quote escapes because of this kind of assumptions. I know this is a long shot, but what if someday packages names do allow spaces? Escapes are virtually free so why not adding them everywhere it's not needed on purpose? Thanks for your time.

**Comment by rmarquis** [comment]
Partially done in 99d03dcf6816b7dc919a60bbf430d98405e385ad. The SC2076warning is in most case a false positive, but the other warnings left can be fixed.

Note: ShellCheck is [available online](http://www.shellcheck.net/) too.

**Comment by Zopieux** [comment]
This is a great progress, thanks and congrats!

**Comment by rmarquis** [comment]
Reverted most of the shellcheck cleanup (array expansion warnings) in da6e5a6 since it causes a regression with pacman flags. I however readded the important fixes in 17ce8c8.

**Comment by rmarquis** [comment]
Closing now. The important bits are now included in the code, and the others bits are either false-positive or not worth including. In particular, "fixing" the space in package names (which will never occur) is error prone, and I even needed to remove some of these parenthesis in 055352bfcc97e54ba9a6763693218e4c9b817008 to fix a bug. I've cleaned up some of these in 6ed7364890d1e2ad785f925adec446da29eb2e32.
